Integrations/integration-EWSv2.yml

```diff
@@ -715,24 +716,39 @@ script:
     def fetch_emails_as_incidents(account, folder_name):
         start_time = EWSDateTime.now(tz=EWSTimeZone.timezone('UTC'))
         last_run = get_last_run()
-        last_emails = fetch_last_emails(account, folder_name, last_run.get(LAST_RUN_TIME), last_run.get(LAST_RUN_IDS))
-
-        ids = []
-        incidents = []
-        for item in last_emails:
-            if item.message_id:
-                ids.append(item.message_id)
-                incident = parse_incident_from_item(item)
-                incidents.append(incident)
-
-        new_last_run = {
-            LAST_RUN_TIME: start_time.ewsformat(),
-            LAST_RUN_FOLDER: folder_name,
-            LAST_RUN_IDS: ids
-        }
 
-        demisto.setLastRun(new_last_run)
-        return incidents
+        try:
+            last_emails = fetch_last_emails(account, folder_name, last_run.get(LAST_RUN_TIME), last_run.get(LAST_RUN_IDS))
+
+            ids = []
+            incidents = []
+            for item in last_emails:
+                if item.message_id:
+                    ids.append(item.message_id)
+                    incident = parse_incident_from_item(item)
+                    incidents.append(incident)
+
+            new_last_run = {
+                LAST_RUN_TIME: start_time.ewsformat(),
+                LAST_RUN_FOLDER: folder_name,
+                LAST_RUN_IDS: ids,
+                ERROR_COUNTER: 0
+            }
+
+            demisto.setLastRun(new_last_run)
+            return incidents
+
+        except RateLimitError:
+            if LAST_RUN_TIME in last_run:
+                last_run[LAST_RUN_TIME] = last_run[LAST_RUN_TIME].ewsformat()
+            demisto.info(str(last_run))
+            if ERROR_COUNTER not in last_run:
+                last_run[ERROR_COUNTER] = 0
+            last_run[ERROR_COUNTER] += 1
+            demisto.setLastRun(last_run)
+            if last_run[ERROR_COUNTER] > 2:
+                raise
```

Comment:
@BenJoParadise please raise an indicative error here such as `EWS encountered more than two errors in fetch....`

Reply:
It is important to raise the exact same error, it is indicative of a configuration, connectivity or authentication error.
Also, raising a different exception will overwrite the stack trace.